### ISSUE TITLE
GATConv does not process None shaped adjecency sparse input

### ISSUE THREAD

**Issue by leotraeg** [danielegrattarola/spektral#462]
Python=3.13.7
Tensorflow=2.20.0
Keras=3.11.3

When calling the functional keras model with GATConv layers with a None shaped adjecency sparse input, I received an error as the GATConv layer multiplies the None mask. This was quite misleading due to the provided examples that do the exact same thing.

```
def pure_gat_models(in_channels=768, hidden_units=128, attn_heads=4, dropout=0.2, decoder="product"):
    # Inputs
    node_embeddings = Input(shape=(in_channels,), name='node_embeddings')  # shape (N, F)
    adjacency_sparse = Input(shape=(None,), sparse=True, name='adjacency_sparse')  # shape (N, N)

    # First GAT layer: multiple attention heads with concatenation and ReLU
    hidden_embeddings = GATConv(
        channels=hidden_units,  attn_heads=attn_heads, concat_heads=True, dropout_rate=dropout, activation='relu'
    )([node_embeddings, adjacency_sparse])

    # Second GAT layer: single attention head, no concat, no activation
    latent_embeddings = GATConv(
        channels=hidden_units,  attn_heads=1, concat_heads=False, dropout_rate=dropout, activation=None
    )([hidden_embeddings, adjacency_sparse])

    # Encoder model
    gat_encoder = Model(inputs=[node_embeddings, adjacency_sparse], outputs=latent_embeddings)

    adjacency_pred = InnerProductDecoder()(latent_embeddings)

    # Combined model
    gat_encoder_decoder = Model(inputs=[node_embeddings, adjacency_sparse], outputs=adjacency_pred)

    return gat_encoder, gat_encoder_decoder
```

My fix was to overwrite the GATConv as follows:

```
class GATConvNoMask(GATConv):
    def call(self, inputs, mask=None):
        # Keras may pass [None, None] for multi-input layers with no masks
        if isinstance(mask, (list, tuple)) and all(m is None for m in mask):
            mask = None
        return super().call(inputs, mask=mask)
```
